python/ml4ir/base/tests/test_lr_schedule.py

```diff
@@ -216,8 +224,86 @@ def test_cyclic_lr_in_training_pipeline(self):
 
         assert np.all(np.isclose(lr_gold, lr_list))
 
-    def tearDown(self):
-        pass
+    def test_reduce_lr_on_plateau_in_training_pipeline(self):
```

Comment:
I see a lot of repetition across all test functions. This should be made cleaner by extracting out the common parts

Reply:
There is only two tests that share some modules, however since these modules are dependent and changes from one test to the other is across different modules, I think there is no very clean way to combine both.